src/gp-angular.js

```diff
@@ -597,6 +598,23 @@ module.directive('gpTranslate', ['$log', 'GlobalizationPipelineService', functio
                     if(DEBUG) {
                         $log.debug("[gp directive] promise resolved...value["+string+"]");
                     }
+
+                    // Check if the translation string has a variable expression
+                    var interpolateFn = $interpolate(string, true);
+                    if (scope.values && interpolateFn) {
+                        try {
+                            // Use the JSON data to populate the variables
+                            var dataContext = JSON.parse(scope.values);

```

Comment:
could `values` already be a map here? Should this be parsed only if `typeof(scope.values) === 'string'`?